spec/services/nomis/booker_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe Nomis::Booker do
+  let(:prisoner) { build_stubbed(:prisoner, nomis_offender_id: 12_345) }
+  let(:lead_visitor) { build_stubbed(:visitor, nomis_id: 1234, sort_index: 0) }
+  let(:additional_visitor) { build_stubbed(:visitor, nomis_id: 2345, sort_index: 1) }
+  let(:banned_visitor) { build_stubbed(:visitor, nomis_id: 2345, banned: true, sort_index: 2) }
+
+  let(:visit) do
+    build_stubbed(:booked_visit,
+      prisoner: prisoner,
+      visitors: [lead_visitor, additional_visitor, banned_visitor])
+  end
+
+  subject { described_class.new(visit) }
+
+  describe '#book' do
+    let(:booking_response) { subject.book }
+
+    describe 'api call to Nomis' do
+      it 'with the correct parameters' do
+        expect(Nomis::Api.instance).
```

Comment:
why not extend the `mock_nomis_with` helper  method to accept an extra argument or block to perform argument call check?